clients/instance/ibm-pi-cloud-connection.go

```diff
@@ -27,7 +26,7 @@ func NewIBMPICloudConnectionClient(ctx context.Context, sess *ibmpisession.IBMPI
 
 // Create a Cloud Connection
 func (f *IBMPICloudConnectionClient) Create(body *models.CloudConnectionCreate) (*models.CloudConnection, *models.CloudConnectionCreateResponse, error) {
-	if strings.Contains(f.session.Options.Zone, helpers.PIStratosRegionPrefix) {
+	if ibmpisession.IsOnPrem(f.session.Options.Zone) {
```

Comment:
Don't you think we should add the func in session struct? This way you do not need to pass the zone and can be called as `f.session.IsOnPrem()`.

In session.go:
```
func (s *IBMPISession) IsOnPrem() bool {
  return strings.Contains(s.Options.Zone, helpers.PIStratosRegionPrefix)
}
```

Reply:
I'm not familiar enough with these objects to say. If you think that's more appropriate, I can do that.